devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy

```diff
@@ -14,181 +13,321 @@
+    if (event) {
+        sendEvent(event)
+    }
+    else {
+        if ((description?.startsWith("catchall:")) || (description?.startsWith("read attr -"))) {
+            def descMap = zigbee.parseDescriptionAsMap(description)
+            if (descMap.clusterInt == 0x0402 && descMap.attrInt == 0) {
+                event = parseTemperature(descMap.value)
+            }
+            else if (descMap.clusterInt == 0x0001 && descMap.attrInt == 0x0020) {
+                event = getBatteryResult(zigbee.convertHexToInt(descMap.value))
+            }
+            else if (descMap.clusterInt == 0x0006 || descMap.clusterInt == 0x0008) {
+                event = parseNonIasButtonMessage(descMap)
+            }
+        }
+        else if (description?.startsWith('temperature: ')) {
+            def value = zigbee.parseHATemperatureValue(description, "temperature: ", getTemperatureScale())
+            event = getTemperatureResult(value)
+        } else if (description?.startsWith('zone status')) {
+            event = parseIasButtonMessage(description)
+        }
 
-def updated() {
-	startTimer()
-    configure()
+        log.debug "Parse returned $event"
+        def result = event ? createEvent(event) : null
+
+        if (description?.startsWith('enroll request')) {
+            List cmds = enrollResponse()
+            result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+        }
+        return result
+    }
 }
 
-def configure(){
-	logIt "Configuring Smart Fob..."
-	[
-	"zdo bind 0x${device.deviceNetworkId} 1 1 6 {${device.zigbeeId}} {}", "delay 200",
-    "zdo bind 0x${device.deviceNetworkId} 2 1 6 {${device.zigbeeId}} {}", "delay 200",
-    "zdo bind 0x${device.deviceNetworkId} 3 1 6 {${device.zigbeeId}} {}", "delay 200",
-    "zdo bind 0x${device.deviceNetworkId} 4 1 6 {${device.zigbeeId}} {}", "delay 200",
-    "zdo bind 0x${device.deviceNetworkId} 1 1 1 {${device.zigbeeId}} {}", "delay 200"
-    ] +
-    zigbee.configureReporting(0x0001,0x0020,0x20,20,20,0x01)
+private Map parseIasButtonMessage(String description) {
+    List parsedMsg = description.split(' ')
+    String msgCode = parsedMsg[2]
+
+    Map resultMap = [:]
+    switch(msgCode) {
+        case '0x0020': // Released
+            resultMap = getButtonResult('release')
+            break
+
+        case '0x0022': // Pressed
+            resultMap = getButtonResult('press')
+            break
+    }
+    return resultMap
 }
 
-def parseCatchAllMessage(descMap) {
-    if (descMap?.clusterId == "0006" && descMap?.command == "01") 		//button pressed
-    	handleButtonPress(descMap.sourceEndpoint as int)
-    else if (descMap?.clusterId == "0006" && descMap?.command == "00") 	//button released
-    	handleButtonRelease(descMap.sourceEndpoint as int)
-    else logIt("Parse: Unhandled message: ${descMap}","trace")
+private Map getBatteryResult(rawValue) {
+    log.debug 'Battery'
+    def volts = rawValue / 10
+    if (volts > 3.0 || volts == 0 || volts == 25.5) {
```

Comment:
For the third check suggest 'rawVolts == 0xFF' because I think it's more clear it's checking for an invalid value for the type.